src/test/java/org/takes/facets/auth/social/PsGithubTest.java

```diff
@@ -0,0 +1,138 @@
+import org.takes.rs.xe.RsXembly;
+import org.takes.rs.xe.XeDirectives;
+import org.xembly.Directives;
+
+/**
+ * Test case for {@link org.takes.rq.RqMethod}.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @since 0.15.2
+ * @todo #42:30min/DEV when task #229 will be completed please add new check
+ *  for the form parameter "code" after line 77. This parameter should
+ *  be equals the variable with name "code", see line 66.
+ */
+public final class PsGithubTest {
+    /**
+     * PsGithub can login.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void canLogin() throws IOException {
+        final String token = "GitHubToken";
+        final String code = "GitHubCode";
+        final int userid = 1;
+        final String login = "octocat";
+        final String avatar = "https://github.com/images/error/octocat.gif";
+        final String lgattr = "login";
+        final String acattr = "access_token";
+        final Take take = new TkFork(
+            new FkRegex(
+                "/login/oauth/access_token",
+                new Take() {
+                    @Override
+                    public Response act(final Request req) throws IOException {
+                        return new RsXembly(
+                            new XeDirectives(
+                                new Directives().add("OAuth")
+                                    .add("token_type").set("bearer").up()
+                                    .add("scope").set("repo,gist").up()
+                                    .add(acattr).set(token).toString()
+                            )
+                        );
+                    }
+                }
+            ),
+            new FkRegex(
+                "/user",
+                new Take() {
+                    @Override
+                    public Response act(final Request req) throws IOException {
+                        MatcherAssert.assertThat(
+                            new RqHref.Base(req).href().param(
+                                acattr
+                            ).iterator().next(),
+                            Matchers.containsString(token)
+                        );
+                        return new RsJSON(
+                            Json.createObjectBuilder()
+                                .add(lgattr, login)
+                                .add("id", userid)
+                                .add("avatar_url", avatar)
+                                .build()
+                        );
+                    }
+                }
+            )
+        );
+        new FtRemote(take).exec(
+            // @checkstyle AnonInnerLengthCheck (100 lines)
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    final Iterator<Identity> itr = new PsGithub(
```

Comment:
@dmzaytsev This variable is only used once. We can get rid of it by calling `next()` directly after `enter()` call.